.github/workflows/build.yml

```diff
@@ -39,7 +39,9 @@ jobs:
         key: ccache-${{ runner.os }}-build-${{ github.sha }}
         restore-keys: ccache-${{ runner.os }}-build-
     - name: install dependencies
-      run: HOMEBREW_NO_AUTO_UPDATE=1 brew install boost hidapi openssl zmq libpgm miniupnpc expat libunwind-headers protobuf ccache
+      run: |
+        HOMEBREW_NO_AUTO_UPDATE=1 brew install boost hidapi openssl zmq libpgm miniupnpc expat libunwind-headers protobuf@21 abseil ccache
```

Comment:
Does this mean every macOS user has to now specifically install and link protobuf@21? This wasn't necessary previously and will break compilation for every Mac user who has the regular protobuf installed.

Reply:
I'm fine with it, as long as it's in `contrib/brew/Brewfile` so MacOS users can build it while following instructions from readme.md

Reply:
> Does this mean every macOS user has to now specifically install and link protobuf@21? This wasn't necessary previously and will break compilation for every Mac user who has the regular protobuf installed.

unfortunately, yes, new protobuf v23 requires newer cmake AFAIK (+some other deps). And we still dont have c++17. Will check what can be done in Brewfile

Reply:
How does it work right now? I use 23 currently to compile monero.

Reply:
I just want to make sure that there is no other way before breaking compilation for all Mac users.

Reply:
macOS has the latest version of CMake installed using brew.

Reply:
@selsta I will try and let you know. When I let CI build osx with Protobuf v23 it failed. There may be another locked dependencies

Reply:
@selsta I am having the same problems as runner in GHA, when upgrading with brew, `make clean ; make debug-test-trezor -j8` ends with an error.

```
  /opt/homebrew/include/absl/utility/utility.h:164:12: error: no member named
  'in_place_t' in namespace 'std'

  using std::in_place_t;
```

and this is c++17 feature https://en.cppreference.com/w/cpp/utility/in_place

So unless monero supports c++17 (why not now?) it has to use old protobuf lib.

Reply:
Maybe there's some macro definition you can supply to disable c++17 features in that code? I can't check right now.

Reply:
@ph4r05 You can add `-DABSL_OPTION_USE_STD_OPTIONAL=0` to cmake command line (or add `add_definitions("-DABSL_OPTION_USE_STD_OPTIONAL=0")` to CMakeLists.txt) to disable C++17 in Abseil.

Reply:
After checking some more, I can see that it should be done by Abseil automatically if you run the compiler with c++14 configured through the command line - `set(CMAKE_CXX_STANDARD 14)` in CMakeLists.txt should work.

Reply:
@SChernykh `try_compile` fails with `"-DCMAKE_CXX_STANDARD=14"`, does not work. any suggestions?

Reply:
I mean, this is no 5 minute fix 😅  so maybe another PR to make it build with the latest protobuf? It is few hours of work, experimenting. Maybe not worth right now

Reply:
This whole protobuf/c++17 thing is not directly related to this PR, it compiled fine when you created this PR. So it can be fixed properly in another PR, I don't mind the current solution here.

Reply:
@SChernykh it does not work with C++14, 
```
/opt/homebrew/include/absl/utility/utility.h:164:12: error: no member named 'in_place_t' in namespace 'std'
using std::in_place_t;
```

absl installation from homebrew has it turned on by default. Also `/opt/homebrew/include/absl/base/options.h`

```
#define ABSL_OPTION_USE_STD_OPTIONAL 1
```

meaning it needs C++17